dynamiq/memory/backend/qdrant.py

```diff
@@ -0,0 +1,146 @@
+import uuid
+
+from qdrant_client.http import models as qdrant_models
+from qdrant_client.http.exceptions import ApiException, UnexpectedResponse
+from qdrant_client.models import FieldCondition, Filter, MatchAny, MatchValue, Range
+
+from dynamiq.components.embedders.base import BaseEmbedder
+from dynamiq.connections import Qdrant as QdrantConnection
+from dynamiq.memory.backend.base import MemoryBackend
+from dynamiq.prompts import Message
+
+
+class QdrantError(Exception):
+    """Base exception for Qdrant-related errors."""
+
+    pass
+
+
+class Qdrant(MemoryBackend):
```

Comment:
@olbychos actually we already have Qdrant and other implementations developed by @MaksymAI. can you check if there is an overlap? I see that there's a major overlap with `dynamiq/storages/vector`

Reply:
Yes, there are some overlaps as well as key differences in the end use cases.
I think we can create a shared Qdrant client later and use "Mixins" to differentiate the backends for memory and RAG.
We can apply the same approach to other vector storage systems, but that can be handled in a separate PR.

Reply:
Okay, but we better do this straight away, so we don't forget it

Reply:
I’ve added a TODO about this:
- We may need to rethink the proper interface for combining the vector store and memory backend( to keep non vector also)
- There will be major code changes, which will mainly affect RAG.  
- We will discuss this with @MaksymAI  to consider how it can be implemented, what is crucial, and to work on the contract.